app/validation/validator.py

```diff
@@ -168,6 +178,35 @@ def validate_routing_on_answer_options(self, block, answer):
                 answer_errors.append(self._error_message(unrouted_error))
         return answer_errors
 
+    @staticmethod
+    def validate_when_rule(when_clause, existing_ids):
+        # Validates any answer id in a when clause exists within the schema
+        when_clause_ids = [when_clause[0]['id']]
```

Comment:
It looks like this is expecting to extract an answer ID from each when clause. But then just creates an array with a single item.

See my comment about iterating through all when rules.